src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -327,7 +341,17 @@ public static bool IsAttributeNameContext(this SyntaxTree syntaxTree, int positi
                 leftToken = syntaxTree.FindTokenOnLeftOfPosition(beforeModifiersPosition, cancellationToken);
                 token = leftToken.GetPreviousTokenIfTouchingWord(beforeModifiersPosition);
 
-                return syntaxTree.IsStatementContext(beforeModifiersPosition, token, cancellationToken);
+                // If one or more attribute lists are present before the caret, check to see if those attribute lists
+                // were written in a local function declaration context.
+                while (token.IsKind(SyntaxKind.CloseBracketToken) && token.Parent.IsKind(SyntaxKind.AttributeList, out AttributeListSyntax attributeList))
+                {
+                    beforeModifiersPosition = attributeList.OpenBracketToken.SpanStart;
+                    leftToken = syntaxTree.FindTokenOnLeftOfPosition(beforeModifiersPosition, cancellationToken);
+                    token = leftToken.GetPreviousTokenIfTouchingWord(beforeModifiersPosition);
+                }
+
+                return syntaxTree.IsStatementContext(beforeModifiersPosition, token, cancellationToken)
+                    || (!syntaxTree.IsScript() && syntaxTree.IsGlobalStatementContext(beforeModifiersPosition, cancellationToken));
```

Comment:
yeah, def not loving the duplication fo the pattern of the IsGlobalSimpleProgramStatementContext  concept.

Reply:
The pull request only shows cases where I needed to add `IsScript` checks. There are a bunch more cases where in doesn't matter.

Reply:
sure.  but i'm not saying we woudl get rid of IsGlobalStatementContext, but more that we would have IsGlobalSimpleProgramStatementContext that we would use in these N locations.

Reply:
i might also rename IsStatementContext because it feels confusing.  maybe IsMethodBodyStatementContext and IsGlobalSimpleProgramStatementContext to help understand the distinction.

Reply:
note: this wasn't a big deal in the past since we barely supported interactive since it was defunded.  but i assume simple-programs will be very important across the IDE, so we should invest in our internals being clearer around hte distinctiion

Reply:
I'm not a fan of the proposed name because it contains `SimpleProgram` as a sub-identifier, which is not the official name for this feature. Do you have an alternative?

Reply:
> I'm not a fan of the proposed name because it contains SimpleProgram as a sub-identifier

That doesn't really bother me.  If this gets renamed we can rename the helper accordingly.

Reply:
I'll file a follow-up issue for this one. Alternatives seem to have as many downsides as current.

Reply:
➡️ #44599